lib/web/apps.go

```diff
@@ -51,12 +52,12 @@ func (h *Handler) clusterAppsGet(w http.ResponseWriter, r *http.Request, p httpr
 		return nil, trace.Wrap(err)
 	}
 
-	resp, err := listResources(clt, r, types.KindAppServer)
+	req, err := convertListResourcesRequest(r, types.KindAppServer)
 	if err != nil {
 		return nil, trace.Wrap(err)
 	}
 
-	appServers, err := types.ResourcesWithLabels(resp.Resources).AsAppServers()
+	appServers, total, err := client.GetResourcePage[types.AppServer](r.Context(), clt, req)
```

Comment:
It looks like the implementation of `client.GetResourcePage` is significantly different than that of `ListResources`. `GetResourcePage` ignores the limit that is passed with `req` and returns all matching resources.